### ISSUE TITLE
icons paint incorrectly on macOS

### ISSUE THREAD

**Issue by lukepighetti** [flutter/flutter#152160]
<img width="442" alt="Screenshot 2024-07-23 at 9 10 22 AM" src="https://github.com/user-attachments/assets/cf906c8b-a07b-4491-9722-6acca60c1c16">

### Steps to reproduce

- create project with sample code below
- run on macOS
- run on iOS

### Expected results

macOS looks like iOS

### Actual results

macOS is not painting correctly. observe the stroke width

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:phosphor_flutter/phosphor_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'zli',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.lightBlue),
      ),
      debugShowCheckedModeBanner: false,
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Column(
          children: <Widget>[
            Row(
              children: [
                IconButton.filledTonal(
                  icon: Icon(PhosphorIconsRegular.user),
                  onPressed: () {},
                ),
                IconButton.filledTonal(
                  icon: Icon(PhosphorIconsRegular.stack),
                  onPressed: () {},
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
```

```yaml
name: zli
description: Get stuff done
publish_to: none
version: 0.1.0+1

environment:
  sdk: ">=3.4.4 <4.0.0"

dependencies:
  flutter:
    sdk: flutter

  cupertino_icons: ^1.0.6
  phosphor_flutter: ^2.1.0

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^3.0.0

flutter:
  uses-material-design: true
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

<img width="442" alt="Screenshot 2024-07-23 at 9 10 22 AM" src="https://github.com/user-attachments/assets/cf906c8b-a07b-4491-9722-6acca60c1c16">

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.3, on macOS 14.4.1 23E224 darwin-arm64, locale en-US)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] VS Code (version 1.91.1)
[✓] Connected device (4 available)
[✓] Network resources

! Doctor found issues in 2 categories.

```

</details>

**Comment by rydmike** [comment]
This is a desktop (macOS, Windows, Linux) Material default adaptive theme response in ThemeData.

If you want it to be none platform adaptive, assign the desired value to it, eg:

materialTapTargetSize: MaterialTapTargetSize.padded,

For the mobile extra padding on all platforms. 

Or the shrinkWrap enum value to have mobile follow the unpadded desktop variant. You probably want the first one though, for same look on desktops as you have on mobile.

**Comment by lukepighetti** [comment]
![image](https://github.com/user-attachments/assets/bae03cb1-cfcb-4457-9340-7f56a221fc9f)

**Comment by rydmike** [comment]
Yeah, sorry, saw it now, I am on a phone, still on the painting issue I see, all I saw was the different paddings ok my phone 🫣

The pixel inaccurate paint on desktops cannot be fixed without major changes to the engine, or so they say. This is yet a lovely sample of what it results in though 👍

**Comment by knopp** [comment]
I don't think this is related to pixel snapping. It renders differently on iPhone 11 (2x scaling) and macOS (also 2x scaling). Maybe a difference in text rendering.

**Comment by knopp** [comment]
FWIS, this is the icon rendered by coretext (no Flutter involved) on macOS: 
<img width="71" alt="Screenshot 2024-07-23 at 18 55 41" src="https://github.com/user-attachments/assets/d647df9f-9051-44b0-b17c-0932a69f6b3c">

So I think this is specific to Phosphor font + coretext on macOS, I don't think it is a bug in Flutter.

**Comment by darshankawar** [comment]
Based on above comment and since the icon is coming from third party pacakage, I'll close this as non-issue from Flutter side. Maybe good to report in dedicated repo: https://github.com/phosphor-icons/flutter/issues
If anyone disagrees, please write in comments and I'll reopen it.

**Comment by bernaferrari** [comment]
I guess your true issue is how hard it is to get icons working on Flutter since there there is no svg support and no one knows how to make them work using fonts.

**Comment by iapicca** [comment]
@darshankawar 
here is a [code sample](https://github.com/iapicca/issue_152160), please reopen the issue

@bernaferrari I don't understand the issue of using icons through fonts (see code sample above)

**Comment by lukepighetti** [comment]
> I guess your true issue is how hard it is to get icons working on Flutter since there there is no svg support and no one knows how to make them work using fonts.

a little off topic but would be nice to see flutter_svg taken on by core team due to recent events

**Comment by bernaferrari** [comment]
The current flutter_sgv is super slow, maybe they can solve with macros, but it is not a solution for now.

The issue is that the only wayto have simple icons are via using custom fonts which is a ton of boot plate to make and maintenance. There is no way he can copy the icon data and run in Flutter.

**Comment by zanderso** [comment]
> a little off topic but would be nice to see flutter_svg taken on by core team due to [recent events](https://github.com/dnfield/flutter_svg/issues/1088)

Still a few details to sort out, but yes the plan is that the core team will maintain flutter_svg.

**Comment by darshankawar** [comment]
> @darshankawar
> here is a [code sample](https://github.com/iapicca/issue_152160), please reopen the issue

Thanks @iapicca 
Maybe since I am not on latest mac + xcode, I don't see the icons being rendered properly:

<img width="130" alt="Screenshot 2024-07-25 at 12 17 17 PM" src="https://github.com/user-attachments/assets/d5598e6c-063d-47fc-b8d8-f9d03d303f9e">

But I will reopen the issue for further insights from the team based on code sample provided.

**Comment by Reprevise** [comment]
> Still a few details to sort out, but yes the plan is that the core team will maintain flutter_svg.

@zanderso Is this still the plan?
_Sorry for the offtopic comment but I'd think it'd be better than opening a new issue about it._

**Comment by zanderso** [comment]
> > Still a few details to sort out, but yes the plan is that the core team will maintain flutter_svg.
> 
> @zanderso Is this still the plan?

Yes.

**Comment by JCKodel** [comment]
Here is another example (Flutter 3.44.0 on MacOs Tahoe 26.5)

<img width="181" height="111" alt="Image" src="https://github.com/user-attachments/assets/3accec2d-1e62-46b0-98b3-201e30bbce35" />

The exactly same icon renders ok in iPhone, Android and Linux (couldn't test Windows)

The source of that icon is https://github.com/phosphor-icons/core/blob/main/raw/regular/file-doc.svg

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256">
  <rect width="256" height="256" fill="none" />
  <path d="M36,152v56H52a28,28,0,0,0,0-56Z" fill="none" stroke="currentColor" stroke-linecap="round"
    stroke-linejoin="round" stroke-width="16" />
  <!-- This is the C -->
  <path d="M216,200.87A22.12,22.12,0,0,1,200,208c-13.26,0-24-12.54-24-28s10.74-28,24-28a22.12,22.12,0,0,1,16,7.13"
    fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="16" />
  <!-- -->
  <path d="M48,112V40a8,8,0,0,1,8-8h96l56,56v24" fill="none" stroke="currentColor" stroke-linecap="round"
    stroke-linejoin="round" stroke-width="16" />
  <polyline points="152 32 152 88 208 88" fill="none" stroke="currentColor" stroke-linecap="round"
    stroke-linejoin="round" stroke-width="16" />
  <ellipse cx="128" cy="180" rx="24" ry="28" fill="none" stroke="currentColor" stroke-linecap="round"
    stroke-linejoin="round" stroke-width="16" />
</svg>
```
